src/internal/session.rs

```diff
@@ -1271,7 +1271,14 @@ mod tests {
     }
 
     #[test]
+    // The counter_mismatch test conforms to the following testing standards:
     // @SF.Messages @TSFI.RESTfulAPI @S0.3
+    //
+    // The test valides encrypting and decrpyting a message, a message order and
+    // duplicate messages. It does so by encrypting messages, then decrypting
+    // them in a different order and asserting the number of messages in between
+    // to be decrypted. Finally, it expects that an attempt to decrypt already
+    // decrypted messages fails.
     fn counter_mismatch() {
```

Comment:
I would go for renaming the function and avoiding the comment altogether, actually.  What about?

```
fn can_decrypt_in_wrong_order_but_can_not_decrypt_twice() {
```

Reply:
I would also appreciate renaming this because the `counter_mismatch` test was one explicitly picked by the certification agency to criticize the naming.

Reply:
Thank you @fisx and @mythsunwind ! The updated name has been introduced in https://github.com/wireapp/proteus/pull/35/commits/2366fd3d259d23866f2ffc91312f05d9b20eca57.